GVRf/Framework/jni/oculus/view_manager.cpp

```diff
@@ -18,41 +18,49 @@
-    std::shared_ptr<RenderTexture> render_texture =
-            *reinterpret_cast<std::shared_ptr<RenderTexture>*>(jrender_texture);
-    std::shared_ptr<ShaderManager> shader_manager =
-            *reinterpret_cast<std::shared_ptr<ShaderManager>*>(jshader_manager);
-    std::shared_ptr<PostEffectShaderManager> post_effect_shader_manager =
-            *reinterpret_cast<std::shared_ptr<PostEffectShaderManager>*>(jpost_effect_shader_manager);
-    std::shared_ptr<RenderTexture> post_effect_render_texture_a =
-            *reinterpret_cast<std::shared_ptr<RenderTexture>*>(jpost_effect_render_texture_a);
-    std::shared_ptr<RenderTexture> post_effect_render_texture_b =
-            *reinterpret_cast<std::shared_ptr<RenderTexture>*>(jpost_effect_render_texture_b);
-
-    activity->viewManager->renderCamera(activity->Scene,
-            scene, camera, render_texture, shader_manager,
-            post_effect_shader_manager, post_effect_render_texture_a,
-            post_effect_render_texture_b,
-            activity->viewManager->mvp_matrix);
+extern "C" {
+
+void Java_org_gearvrf_GVRViewManager_renderCamera(JNIEnv * jni, jclass clazz,
+		jlong appPtr, jlong jscene, jlong jcamera, jlong jrender_texture,
+		jlong jshader_manager, jlong jpost_effect_shader_manager,
+		jlong jpost_effect_render_texture_a,
+		jlong jpost_effect_render_texture_b) {
+	GVRActivity *activity = (GVRActivity*) ((App *) appPtr)->GetAppInterface();
+
+	std::shared_ptr<Scene> scene =
+			*reinterpret_cast<std::shared_ptr<Scene>*>(jscene);
+	std::shared_ptr<Camera> camera =
+			*reinterpret_cast<std::shared_ptr<Camera>*>(jcamera);
+	std::shared_ptr<RenderTexture> render_texture =
+			*reinterpret_cast<std::shared_ptr<RenderTexture>*>(jrender_texture);
+	std::shared_ptr<ShaderManager> shader_manager =
+			*reinterpret_cast<std::shared_ptr<ShaderManager>*>(jshader_manager);
+	std::shared_ptr<PostEffectShaderManager> post_effect_shader_manager =
+			*reinterpret_cast<std::shared_ptr<PostEffectShaderManager>*>(jpost_effect_shader_manager);
+	std::shared_ptr<RenderTexture> post_effect_render_texture_a =
+			*reinterpret_cast<std::shared_ptr<RenderTexture>*>(jpost_effect_render_texture_a);
+	std::shared_ptr<RenderTexture> post_effect_render_texture_b =
+			*reinterpret_cast<std::shared_ptr<RenderTexture>*>(jpost_effect_render_texture_b);
+
+	activity->viewManager->renderCamera(activity->Scene, scene, camera,
+			render_texture, shader_manager, post_effect_shader_manager,
+			post_effect_render_texture_a, post_effect_render_texture_b,
+			activity->viewManager->mvp_matrix);
+}
+
+void Java_org_gearvrf_GVRViewManager_readRenderResult(JNIEnv * jni, jclass clazz,
+		jlong jrender_texture, jobject jreadback_buffer) {
+
+	uint8_t *pReadbackBuffer = (uint8_t *) jni->GetDirectBufferAddress(
+			jreadback_buffer);
+	std::shared_ptr<RenderTexture> render_texture =
+			*reinterpret_cast<std::shared_ptr<RenderTexture>*>(jrender_texture);
+	int mWidth = render_texture->width();
```

Comment:
The 'm' prefix means this is a "member variable" - to the extent that it has any point at all, it is that it is NOT local; NOT declared nearby, but in the class declaration.

Please use "width" and "height".